lib/ecto/adapters/mysql/connection.ex

```diff
@@ -714,6 +714,19 @@ if Code.ensure_loaded?(Mariaex) do
     @impl true
     def ddl_logs(_), do: []
 
+    @impl true
+    def table_exists_query(table) do
+      require Ecto.Query
```

Comment:
Ironically, now that we moved it here, we can write it as a raw string, since we have all of the escape helpers available. Haha, but no problem. I can do that in separate commit.

Reply:
LOL 😅

Reply:
But I think this was a very good example on how the hardest part of the problem, in this case, is finding the proper contract between the different layers. Once the contract is found, then it is just a matter of changing the implementation. :)

Reply:
True! ✨ (off topic: I learned a lot about how ecto works in this process)